text/0088-metric-instrument-optional-refinements.md

```diff
@@ -0,0 +1,205 @@
+calling patterns as the foundational instrument they refine, adding
+either a different standard implementation or a restriction of the
+input domain.
+
+We have done away with instrument options, in other words, in favor of
+optional metric instruments.  Here we discuss three important
+capabilities achievable using instrument refinements.
+
+#### Non-negative
+
+For some instruments, such as those that measure real quantities,
+negative values are meaningless.  For example, it is impossible for a
+person to weigh a negative amount.
+
+A non-negative instrument refinement accepts only non-negative values.
+For instruments with this property, negative values are considered
+measurement errors.
+
+#### Monotonic
+
+A monotonic instrument is one where the user promises that successive
+metric events for a given instrument definition and label set will
+differ by a non-negative value.  This is defined in terms of the last
+value relationship, therefore only applies to refinements of the
+Observer instrument. For example, the CPU time used by a process as
+read in successive collection intervals cannot change by a negative
+amount, because it is impossible to use a negative amount of CPU time.
+
+A monotonic instrument refinement accepts only values that are greater
+than or equal to the last-captured value of the instrument, for a
+given label set.  For instruments with this property, values that are
+less than some prior value are considered measurement errors.
+
+#### Sum-Only
+
+A sum-only instrument is one where only the sum is considered to be of
+interest.  For a Sum-Only instrument refinement, we have a semantic
+property that two events with numeric values `M` and `N` are
+semantically equivalent to a single event with value `M+N`.  For
+example, in a count of users arriving by bus to an event, we are not
+concerned with the number of buses that arrived.
+
+A sum-only instrument is one where the number of events is not
+counted, by default, only the `Sum`.
+
+#### Language-level refinements
+
+OpenTelemetry implementations may wish to add instrument refinements
+to accommodate built-in types.  Languages with distinct integer and
+floating point should offer instrument refinements for each, leading
+to type names like `Int64Measure` and `Float64Measure`.
+
+A language with support for unsigned integer types may wish to create
+dedicated instruments to report these values, leading to type names
+like `UnsignedInt64Observer` and `UnsignedFloat64Observer`.  These
+would naturally apply a non-negative refinment.
+
+Other uses for built-in type refinements involve the type for duration
+measurements.  For example, where there is built-in type for the
+difference between two clock measurements, OpenTelemetry APIs should
+offer a refinement to automatically apply the correct unit of time to
+the measurement.
+
+### Counter refinement
+
+Counter is a non-negative, sum-only refinement of the Measure
+instrument.
```

Comment:
Counters can't count? 🙃

Reply:
I.e., Counters don't count in the default aggregation.